### PR TITLE
allow creating and removing PayPal orders for multiple posts and tickets

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/97806

### CODE REVIEW

File: src/Commerce/Generator/PayPal/CLI.php [hunk 1]
```diff
@@ -50,24 +45,63 @@ public function generate_orders( array $args = array(), array $assoc_args = arra
...
+					WP_CLI::error( "Ticket with ID {$ticket_id} is related to a post with ID {$post_id} but that post does not exist, please edit the ticket and retry." );
+				}
+
+				$ticket = $paypal->get_ticket( $post_id, $ticket_id );
+
+				$this->backup_ticket_total_sales( $ticket_id );
+
+				$inventory  = $ticket->inventory();
+				$inventory  = - 1 === $inventory ? 'unlimited' : $inventory;
+				$this_ticket_qty = 'unlimited' === $inventory ? $this_ticket_qty : min( $this_ticket_qty, (int) $inventory );
```
Comment (barryhughes) [inline src/Commerce/Generator/PayPal/CLI.php]: Hmm ... why not kill the previous line and do `- 1 === $inventory` for the ternary condition on this line?

File: src/Commerce/Generator/PayPal/CLI.php [hunk 2]
```diff
@@ -414,25 +440,47 @@ protected function generate_tickets( $order_status, $transaction_data ) {
...
+
 		$pre_deleted_attendees_count = (int) get_post_meta( $post_id, '_tribe_deleted_attendees_count', true );
 
-		WP_CLI::log( "Removing generated PayPal orders from post {$post_id}" );
+		/** @var \Tribe__Tickets__Commerce__PayPal__Main $paypal */
+		$paypal = tribe( 'tickets.commerce.paypal' );
 
-		$orders = Order::find_by( array( 'post_id' => $post_id, 'posts_per_page' => - 1 ) );
+		$is_ticket = true;
+		if (!$post->post_type === $paypal->ticket_object) {
```
Comment (barryhughes) [inline src/Commerce/Generator/PayPal/CLI.php]: Spacing inside the parentheses / around the `!` operator

File: src/Commerce/Generator/PayPal/CLI.php [hunk 3]
```diff
@@ -442,16 +490,26 @@ public function reset_orders( array $args = array(), array $assoc_args = array()
...
-			if ( ! in_array( $post_id, $post_ids ) ) {
-				continue;
-			}
+			$attendees = $order->get_attendees();
 
-			if ( 1 === ! count( $post_ids ) ) {
-				WP_CLI::error( 'This tool does not support creating or deleting orders for multiple tickets or posts (yet).' );
+			if ( $is_ticket ) {
+				$attendees = array_filter( $attendees, function ( array $attendee ) use ( $ticket_id ) {
+					return isset( $attendee['product_id'] ) &&  $attendee['product_id']==$ticket_id;
```
Comment (barryhughes) [inline src/Commerce/Generator/PayPal/CLI.php]: Spacing (double space after `&&` and no space either side of `==`)

File: src/Commerce/Generator/PayPal/CLI.php [hunk 4]
```diff
@@ -442,16 +490,26 @@ public function reset_orders( array $args = array(), array $assoc_args = array()
...
+				} );
+			} else {
+				$attendees = array_filter( $attendees, function ( array $attendee ) use ( $post_ids ) {
+					return isset( $attendee['event_id'] ) && in_array( $attendee['event_id'], $post_ids );
+				} );
 			}
 
-			$attendees = $order->get_attendees();
-			$order->delete( true, true );
+			if (count($post_ids) === 1) {
```
Comment (barryhughes) [inline src/Commerce/Generator/PayPal/CLI.php]: Spacing

**Review by barryhughes — APPROVED** [review]
Looks good! Left some notes, none are dealbreakers

**Comment by lucatume** [comment]
Thanks @barryhughes!

**Pull request merged**